src/core/backend_context.cc

```diff
@@ -291,24 +295,56 @@ BackendResponder::Finalize()
                 status),
             "error sending TensorFlow response");
       } else {
-        bool cuda_used = false;
-        status = CopyBuffer(
-            response_output->Name(), pinned_memory_type, pinned_memory_id,
-            response_memory_type, response_memory_type_id, response_byte_size,
-            pinned_buffer + offset, const_cast<void*>(response_buffer), stream_,
-            &cuda_used);
-        need_sync_ |= cuda_used;
+        // Only use parallel CopyBuffer for CPU-CPU copy when worker count > 1
+        if ((AsyncWorkQueue::GetWorkerCount() > 1) &&
```

Comment:
You should have a "bool use_async_cpu_copy" that is set when the class is constructed.

Reply:
Do you mean as a member of the BackendInputCollector and BackendResponder class?